threatgrid/ctia
# Add route to link Incidents to Investigations

Close https://github.com/threatgrid/iroh/issues/2431

Extends `/incident/:id/link` to also support linking to Investigations.





<a name="qa">[§](#qa)</a> QA
============================

This PR broadens the scope of the `/incident/:id/link` route. For a body, it now accepts *one of* an `investigation_id` or `casebook_id` field, as well as some other optional keys (at the moment, just `tlp`).

Please follow these steps to create a relationship between an incident and an investigation, and verify it was created.

1. Visit `https://private.intel.int.iroh.site/index.html#/Incident/post_ctia_incident__id__link` (or equivalent)
2. Add a known incident id to the `id` param.
3. Add a body of the form `{ "investigation_id" : "<known-investigation-id>", ...}` with any relevant optional keys, then execute.
4. The response should be `200` and contain an `id` field. This is the relationship that was created.
5. Verify the `id` returned in step 4 corresponds to a relationship from the incident id to the investigation id.

The previous step should also still work for linking casebooks, via the `casebook_id` field. This behavior should be unchanged.

Please also verify providing zero or more than one of `investigation_id` or `casebook_id` fields in the body of the `/incident/:id/link` route triggers a 400 error response. Error messages should also be descriptive enough to diagnose the problem immediately.






<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================



```
intern: Add route to link Incidents to Investigations
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
